version: '3'

services:
  app:
    build:
      context: ./node-app
    image: acafons79/node-app
    container_name: app
    volumes:
      - ./node-app:/home/node/app
    tty: true
    expose:
      - "3000"
    depends_on:
        db:
          condition: service_healthy
    networks:
      - node-network

  db:
    image: mysql:9.0.1
    command: --innodb-use-native-aio=0
    container_name: db
    restart: always
    tty: true
    expose:
      - "3306"
    volumes:
      - ./mysql/mysql-files:/var/lib/mysql
      - ./mysql/tables:/docker-entrypoint-initdb.d
    environment:
      - MYSQL_ROOT_PASSWORD=root
      - MYSQL_DATABASE=nodedb
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 10s
      timeout: 10s
      start_period: 10s
      retries: 3
    networks:
      - node-network

  nginx:
    build:
      context: ./nginx
    image: acafons79/node-nginx
    container_name: nginx
    networks:
      - node-network
    ports:
      - "8080:80"
    depends_on:
      - app

networks:
  node-network:
    driver: bridge
